src/openrct2/world/Park.cpp

```diff
@@ -561,32 +561,29 @@ uint32_t Park::CalculateSuggestedMaxGuests() const
 
         // Add guest score for ride type
         suggestedMaxGuests += ride.GetRideTypeDescriptor().BonusValue;
+
+        // If difficult guest generation, extra guests are available for good rides
+        if (!(gParkFlags & PARK_FLAGS_DIFFICULT_GUEST_GENERATION))
```

Comment:
Since this check is unlike the others, I’d prefer if it didn’t return early but instead encompassed the others. That would also help legibility.